### PR TITLE
#211: getLogo and getLogoP methods

### PR DESCRIPTION
**Pull request by BeeMargarida**
| - | - |
| --- | --- |
| Issue | #211 |
| Dependencies | -- |
| Decisions | getLogoP and getLogo methods |
| Animated GIF | -- |

### CODE REVIEW

File: src/js/api/brand.js [hunk 1]
```diff
@@ -11,6 +11,45 @@ if (
...
+ * @param {Object} options A map with options, such as:
+ *  - 'brand' - The brand of the model
+ *  - 'version' - The version of the build, defaults to latest
+ *  - 'variant' - The variant of the logo, that controls semantics of the logo
+ *  - 'format' - The format of the logo image to be retrieved defaults to png
+ *  - 'size' - The size of the logo image
+ * @param {Function} callback Function with the result of the request.
+ * @returns {XMLHttpRequest} The brand's logo.
+ */
+ripe.Ripe.prototype.getLogo = function(options, callback) {
```
Comment (gcandal) [inline src/js/api/brand.js:26]: Add an equivalent to `getMeshUrl` as well, as it allows more flexible use-cases.

File: src/js/api/brand.js [hunk 2]
```diff
@@ -11,6 +11,45 @@ if (
...
 }
 
+/**
+ * Returns the logo of a brand.
+ *
+ * @param {Object} options A map with options, such as:
+ *  - 'brand' - The brand of the model
+ *  - 'version' - The version of the build, defaults to latest
+ *  - 'variant' - The variant of the logo, that controls semantics of the logo
+ *  - 'format' - The format of the logo image to be retrieved defaults to png
```
Comment (gcandal) [inline src/js/api/brand.js:21]:
```suggestion
 *  - 'format' - The format of the logo image to be retrieved (defaults to 'png')

```

File: src/js/api/brand.js [hunk 3]
```diff
@@ -11,6 +11,45 @@ if (
...
+ *  - 'size' - The size of the logo image
+ * @returns {XMLHttpRequest} The brand's logo.
```
Comment (gcandal) [inline src/js/api/brand.js:43]:
```suggestion
 * @returns {Promise} The brand's logo.
```

**Review by gcandal — CHANGES_REQUESTED** [review]

File: src/js/api/brand.js (hunk 1 above)
Comment (gcandal) [reply src/js/api/brand.js:26]: Nevermind, I just noticed it already exists.

**Comment by gcandal** [comment]
@BeeMargarida I've just noticed we have no [documentation](https://docs.platforme.com/) for the brand endpoints. Can you take care of that?

File: test/js/api/brand.js [hunk 4]
```diff
@@ -28,6 +28,21 @@ describe("BrandAPI", function() {
         });
     });
 
+    describe("#getLogoP()", function() {
```
Comment (gcandal) [inline test/js/api/brand.js:31]: Move this above the `getLogoUrl` test, as you have done for `src/js/api/brand.js`

**Review by gcandal — CHANGES_REQUESTED** [review]

**Review by gcandal — APPROVED** [review]

**Review by joamag — APPROVED** [review]

**Pull request merged**